File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left = 0;
        int right = heights.size() - 1;
        int max_water = 0;
        while (left < right) {
            int result = (right - left) * std::min(heights[left], heights[right]);
            max_water = std::max(max_water, result); 
            if (heights[left] < heights[right]) {
                left++;
            } else {
                right--;
            } 
        }

        return max_water;
    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        return self.dfs(root, 0)

    def dfs(self, node, depth):
        if not node:
            return depth
        
        left = self.dfs(node.left, depth + 1)
        right = self.dfs(node.right, depth + 1)

        return max(left, right)
